Add breadcrumbs to cli/commands.

StevenDufresne opened the pull request:
This PR adds the breadcrumbs to `cli/commands`. 

I realized after logging an issue that the [designs](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?type=design&node-id=3745-10313&mode=design) don't have the breadcrumb. However, I don't understand why this page would not have it when the block-editor handbook _landing page_ and other landing pages have it. 
It's also linked in deeper nested pages as the parent and last [production](https://developer.wordpress.org/cli/commands/) currently has the breadcrumb there.

## Screenshot
![localhost_8888_cli_commands_ (1)](https://github.com/WordPress/wporg-developer/assets/1657336/2ab7ccef-80b8-4e70-a911-342db6a1bfdb)

## Nested Page
![localhost_8888_cli_commands_admin_](https://github.com/WordPress/wporg-developer/assets/1657336/4aa3eab4-74d4-4449-8b0a-bdf1b8333716)

StevenDufresne committed:
Add breadcrumbs to cli.

StevenDufresne commented:
@ndiego Thoughts?

adamwoodnz commented:
If we do add this we'll need to adjust the vertical position of the ToC; it should align with the title.

StevenDufresne commented:
I'm getting no table of contents on the [staging env](https://developer.wordpress.org/redesign-test/cli/commands/) (well, i'm getting the back to top).

<img width="1292" alt="Screenshot 2023-11-06 at 2 59 39 PM" src="https://github.com/WordPress/wporg-developer/assets/1657336/5901f491-c7e2-4681-9298-3321affb1f33">

StevenDufresne commented:
Also, if we implement https://github.com/WordPress/wporg-developer/issues/312, then I imagine it will just work...?

adamwoodnz commented:
> Also, if we implement #312, then I imagine it will just work...?

That search bar with bottom border is only for handbook pages if I understand it correctly.

fcoveram commented:
> …I don't understand why this page would not have it when the block-editor handbook landing page and other landing pages have it.

Breadcrumbs appear from the third level and beyond (Home / Level 2 / Level 3). The idea was to avoid duplicative content as in level 2's pages local nav shows the current page in bold. So in the [nested page example](https://user-images.githubusercontent.com/1657336/280603800-4aa3eab4-74d4-4449-8b0a-bdf1b8333716.png), it's correct to show it.

> That search bar with bottom border is only for handbook pages if I understand it correctly.

~~Yes~~. ← Updating this response as [this comment](https://github.com/WordPress/wporg-developer/issues/341#issuecomment-1794462249) informs about an update of this component. The bottom border in Articles should be removed.

adamwoodnz commented:
> So in the [nested page example](https://user-images.githubusercontent.com/1657336/280603800-4aa3eab4-74d4-4449-8b0a-bdf1b8333716.png), it's correct to show it.

Ah ok, so I think we only need to add this to single-command.html

StevenDufresne commented:
> Breadcrumbs appear from the third level and beyond (Home / Level 2 / Level 3). The idea was to avoid duplicative content as in level 2's pages local nav shows the current page in bold. So in the [nested page example](https://user-images.githubusercontent.com/1657336/280603800-4aa3eab4-74d4-4449-8b0a-bdf1b8333716.png), it's correct to show it.

@fcoveram Does that mean it shouldn't be in the [block editor handbook](https://developer.wordpress.org/redesign-test/block-editor/)?

<img width="1547" alt="Screenshot 2023-11-07 at 2 22 54 PM" src="https://github.com/WordPress/wporg-developer/assets/1657336/b14aa9e6-3c49-44c3-9df4-e98f11dae7f7">

fcoveram commented:
> @fcoveram Does that mean it shouldn't be in the [block editor handbook](https://developer.wordpress.org/redesign-test/block-editor/)?

Exactly.

To assess the change impact, which site sections will have this change? I'm thinking of Developers, Showcase, About WordPress, and News.

The idea of showing it from level 3 is to diminish location redundancy. But in the case of Articles, it feels weird landing on its main page without breadcrumb and then seeing it in all internal pages.

outdoor2kode commented:
Summarizing the breadcrumb behaviors I've observed while working on another ticket.

**Handbook** ✅
Only put two here. All handbooks have the same look.
<img width="158" alt="image" src="https://github.com/WordPress/wporg-developer/assets/18050944/73f61aca-826a-4bfb-919b-7c44d3eeed4e">

| level2 | level3 |
|-|-|
| <img width="1682" alt="Screenshot 2023-11-10 at 12 42 30 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/48cbf9f0-0ed2-4c42-8976-7225bbc334cc"> | <img width="1670" alt="Screenshot 2023-11-10 at 12 42 37 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/9e1c030d-7e70-433a-a65a-1b4489d7eece"> | 
| <img width="1646" alt="Screenshot 2023-11-10 at 12 43 21 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/18f67ddf-0822-4b38-b0e1-edb03e6b7f39"> | <img width="1591" alt="Screenshot 2023-11-10 at 12 43 28 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/e1c08dfb-c526-4630-bc1c-de4ddeec120b"> |

**Resource** ✅
If clicking on Resource in the breadcrumbs, it would redirect back to Dashicons.
Same in production, visiting Resource page would redirect to Dashicons page.
And there's no breadcrumbs on production, but there's a title though.

| local/staging | production |
|-|-|
| <img width="1635" alt="Screenshot 2023-11-10 at 12 45 28 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/eef79d58-8139-4b2c-8c23-60edb281ddba"> | <img width="988" alt="Screenshot 2023-11-10 at 12 45 35 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/c16e65d8-5859-4b1a-aec5-4ab0f9fa7d0b"> |

**Reference** ✅

| local/staging | production |
|-|-|
| <img width="1620" alt="Screenshot 2023-11-10 at 12 46 02 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/13972ac4-ea4d-4b8d-ad7c-39a44bed2deb"> | <img width="1145" alt="Screenshot 2023-11-10 at 12 46 08 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/b00763bd-0b89-44cb-b58d-9ffb8ceabc07"> |

**Functions**  ✅
Only put Functions CPT here. Other CPTs have the same look.
<img width="158" alt="image" src="https://github.com/WordPress/wporg-developer/assets/18050944/44fd875a-22f8-4c5b-afa1-7373fbb1cb29">

In production, it has 4 levels.
We don't need the 'Reference' level? => **Done Adding back.**

| | level2 | level3 |
|-|-|-|
| local/staging | <img width="1625" alt="Screenshot 2023-11-10 at 12 46 58 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/9f38311c-ccaa-4517-8271-323aacd9effb"> | <img width="1550" alt="Screenshot 2023-11-10 at 12 47 25 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/4e48a07d-2df3-45b9-83be-5f67ee33dc77"> |
| production | <img width="752" alt="Screenshot 2023-11-10 at 12 47 06 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/facba1c7-123c-4e75-b7ed-4f231717f41a"> | <img width="974" alt="Screenshot 2023-11-10 at 1 31 58 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/41166a9a-2d3a-4163-bfd7-f4776d96374b"> |

**Taxonomy** ✅
There are overall three taxonomies - since, packages, files
<img width="327" alt="image" src="https://github.com/WordPress/wporg-developer/assets/18050944/faecbb5e-c205-460b-ac2e-f8cf107423d5">

Same here, in production, it has 4 levels.
We don't need the 'Reference' level? =>  **Done Adding back.** ✅

| local/staging, files | production |
|-|-|
| <img width="1627" alt="Screenshot 2023-11-10 at 12 47 59 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/50a59b61-ed17-4cec-99e7-a10457adfc9f"> | <img width="919" alt="Screenshot 2023-11-10 at 12 48 07 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/f2cb1d8c-d954-4b73-8891-f49b61bc2c60"> |

There's breadcrumbs on packages page, but isn't on since page. => **Done Adding back.** ✅

| packages | since |
|-|-|
| <img width="1343" alt="Screenshot 2023-11-10 at 12 48 21 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/6e80526f-c51f-48d2-9f6b-12bfc08ba264"> | <img width="1278" alt="Screenshot 2023-11-10 at 1 45 06 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/38e8a3c4-b7d6-4a72-8855-514b3fb6fc07"> |

The files and since pages can be accessed through the links as indicated below.
Not sure how to access the packages page.

<img width="824" alt="image" src="https://github.com/WordPress/wporg-developer/assets/18050944/24383a45-337d-451f-9961-3ead123b7eaf">

**CLI/Commands** ✅

I guess this is the most standard one in terms of design.
It only renders breadcrumbs when it's level 3.
It also makes sense not to include Reference in breadcrumbs on CLI/Commands page because there's a button on homepage linking to the page.

| | level2 | level3 |
|-|-|-|
| local/staging | <img width="1384" alt="Screenshot 2023-11-10 at 12 54 49 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/cbcef9a0-33da-4347-a274-434f06aa73f0"> | <img width="1256" alt="Screenshot 2023-11-10 at 12 55 07 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/d84b294b-ca46-47a4-bd12-8c4511817881"> |
| production | <img width="1088" alt="Screenshot 2023-11-10 at 12 54 58 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/e926666f-e20c-4b7a-a495-f8d40c0c8b8c"> | <img width="996" alt="Screenshot 2023-11-10 at 12 55 11 AM" src="https://github.com/WordPress/wporg-developer/assets/18050944/190241b0-d4e4-46e5-a760-6ccdff343604"> |

outdoor2kode commented:
Currently, most of the pages on developer sites display breadcrumbs at 2 levels. 
And to be more specific, [CLI-Commands page](https://developer.wordpress.org/redesign-test/cli/commands/) is the only one that doesn’t display at 2 levels.

And after chatting with @fcoveram, we agree that it makes sense to display the breadcrumbs from `Home / Level 1`. This is also what we have in Showcase for the site details pages.

Note: In our discussion, it was mentioned that how we describe the levels is slightly off. 
All the comments above are labeled as `Home / Level 2 / Level 3`, but they should actually be `Home / Level 1 / Level 2`.

outdoor2kode committed:
Add breadcrumbs to reference.

outdoor2kode committed:
Add back reference to breadcrumbs

outdoor2kode committed:
Add breadcrumbs to since.

outdoor2kode commented:
@adamwoodnz 

I have already added the pages missing the `Home/Level1` breadcrumbs. 
To test, you can refer to the comments above, where I have marked them with ✅.

Additionally, I have temporarily re-added the 'reference' to the breadcrumbs for now. 
Without 'reference', the process becomes a bit odd, as users won't be able to navigate back to the reference page from any CPT pages. 

![Screenshot 2023-11-29 at 1 37 46 AM](https://github.com/WordPress/wporg-developer/assets/18050944/b9e963c0-97aa-466b-aebb-ff3eb5be2c76)

~However, [this PR](https://github.com/WordPress/wporg-developer/pull/166) mentions the original reason for its removal. ([Ref](https://github.com/WordPress/wporg-developer/pull/166/files#diff-1cf9dd600383a4c89ba09a878ca8691359b6f6dcece2b9c9b90f3de810f274a4R234))~

~@StevenDufresne, is this still a concern? Is the reason we need to remove the 'reference' from the breadcrumbs due to the i18n issue, such as the breadcrumbs being too long?~

**Update**
According to Steve's feedback,
It was removed because the page was gone.
Now it can be brought back.

fcoveram commented:
Thanks @renintw for addressing this.

> The idea was to avoid duplicative content as in level 1’s pages local nav shows the current page in bold.

That statement shared by me days ago is slightly incorrect. There are L1 pages not displayed in the local nav menu, so displaying them is safer than not. That's the rationale behind this decision.

adamwoodnz commented on source/wp-content/themes/wporg-developer-2023/templates/archive-command.html:
```diff
@@ -1,4 +1,4 @@
-<!-- wp:template-part {"slug":"header-no-breadcrumbs","className":"has-display-contents"} /-->
```
Looks like this template part `header-no-breadcrumbs.html` is unused after these changes and can be removed.

Also looks like the `local-nav.html` template part, which was used in both header parts, can be moved inline within `header.html` and the template part removed.

outdoor2kode replied:
Thanks. Done!

adamwoodnz reviewed (changes requested):
Nice work, I like how this simplifies the template setup.

A couple of comments inline.

One more adjustment required now that the CLI Command archive has breadcrumbs; we need to remove this custom ToC spacing rule: https://github.com/WordPress/wporg-developer/blob/trunk/source/wp-content/themes/wporg-developer-2023/src/style/style.scss#L407

Removing that should fix the misaligned ToC and H1:

![Screenshot 2023-11-30 at 10 06 48 AM](https://github.com/WordPress/wporg-developer/assets/1017872/c73700df-b196-4a86-a0c4-9bcc8b29e10a)

outdoor2kode committed:
Remove header-no-breadcrumbs.html

outdoor2kode committed:
Moved local-nav inline within header.html & removed it.

outdoor2kode committed:
Remove extra TOC margin top.

Since the CLI Command archive has breadcrumbs now,
The misaligned ToC and H1 can be fixed through removing it.

outdoor2kode commented:
## Updated Screenshot

![image](https://github.com/WordPress/wporg-developer/assets/18050944/b8cedfed-07c2-454f-9583-57f8fe2bcff0)

adamwoodnz reviewed (approved):
LGTM :shipit:
